File: programmers/152995.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

typedef pair<int, int> P;

int n, a, b;
vector<int> v;

int solution(vector<vector<int>> scores) {
    int answer = 0;
    
    n = scores.size();
    a = scores[0][0];
    b = scores[0][1];
    sort(scores.begin(), scores.end(), greater());
    
    v.push_back(scores[0][0] + scores[0][1]);
    int max = 0, nmax= scores[0][1];
    for (int i = 1; i < n; i++) {
        if (scores[i - 1][0] > scores[i][0]) {
            if (max < nmax) max = nmax;
            nmax = scores[i][1];
        }
        if (nmax < scores[i][1]) nmax = scores[i][1];
        if (max > scores[i][1]) {
            if (scores[i][0] == a && scores[i][1] == b) answer = -1;
            continue;
        }
        v.push_back(scores[i][0] + scores[i][1]);
    }
    
    if (answer > -1) {
        sort(v.begin(), v.end(), greater());
        for (int i = 0; i < v.size(); i++) {
            if (v[i] == a + b) {
                answer = i + 1;
                break;
            }
        }
    }
    
    return answer;
}

File: programmers/152995.java
import java.util.*;

class Solution {
    public int solution(int[][] scores) {
        int n = scores.length;
        int[] target = scores[0];
        
        // 첫 번째 점수 기준으로 내림차순 정렬
        Arrays.sort(scores, (s1, s2) -> Integer.compare(s2[0], s1[0]));
        
        int max = 0;
        int tmp = scores[0][1];
        
        List<Integer> sums = new ArrayList<>();
        sums.add(scores[0][0] + scores[0][1]);
        
        // 인센티브 지급 대상 추리기
        for (int i = 1; i < n; i++) {
            // 최대값 갱신
            if (scores[i][0] < scores[i - 1][0]) {
                max = Math.max(max, tmp);
                tmp = scores[i][1];
            }
            tmp = Math.max(tmp, scores[i][1]);
            
            // 탈락 조건
            if (max > scores[i][1]) {
                if (scores[i][0] == target[0] && scores[i][1] == target[1]) {
                    return -1;
                }
                continue;
            }
            
            sums.add(scores[i][0] + scores[i][1]);
        }
        
        // 내림차순 정렬
        sums.sort((a, b) -> Integer.compare(b, a));
        
        // 순위 계산
        int sum = target[0] + target[1];
        for (int i = 0; i < sums.size(); i++) {
            if (sums.get(i) == sum) {
                return i + 1;
            }
        }
        return -1;
    }
}